coverage_point_calculator/src/lib.rs

```diff
@@ -0,0 +1,933 @@
+/// [boosted-hex-restriction]: https://github.com/helium/oracles/pull/808
+///
+use crate::{
+    hexes::{CoveredHex, CoveredHexes},
+    location::{LocationTrust, LocationTrustScores},
+    speedtest::Speedtest,
+};
+use coverage_map::{RankedCoverage, SignalLevel};
+use rust_decimal::{Decimal, RoundingStrategy};
+use rust_decimal_macros::dec;
+use speedtest::Speedtests;
+
+pub mod hexes;
+pub mod location;
+pub mod speedtest;
+
+pub type Result<T = ()> = std::result::Result<T, Error>;
+
+/// Necessary checks for calculating coverage points is done during [RewardableRadio::new].
+#[derive(Debug, Clone)]
+pub struct RewardableRadio {
+    radio_type: RadioType,
+    speedtests: Speedtests,
+    location_trust_scores: LocationTrustScores,
+    radio_threshold: RadioThreshold,
+    covered_hexes: CoveredHexes,
+    boosted_hex_eligibility: BoostedHexStatus,
+}
+
+/// Output of calculating coverage points for a [RewardableRadio].
+///
+/// The only data included was used for calculating coverage points.
+///
+/// - If more than the allowed speedtests were provided, only the speedtests
+///   considered are included here.
+///
+/// - When a radio covers boosted hexes, [location_trust_scores] will contain a
+///   trust score _after_ the boosted hex restriction has been applied.
+///
+/// - When a radio is not eligible for boosted hex rewards, [covered_hexes] will
+///   have no boosted_multiplier values.
+///
+#[derive(Debug)]
+pub struct CoveragePoints {
+    /// Value used when calculating poc_reward
+    pub total_coverage_points: Decimal,
+    /// Coverage Points collected from each Covered Hex
+    pub hex_coverage_points: Decimal,
+    /// Location Trust Multiplier, maximum of 1
+    pub location_trust_multiplier: Decimal,
+    /// Speedtest Mulitplier, maximum of 1
+    pub speedtest_multiplier: Decimal,
+    // ---
+    pub radio_type: RadioType,
+    pub radio_threshold: RadioThreshold,
+    pub speedtests: Vec<Speedtest>,
+    pub location_trust_scores: Vec<LocationTrust>,
+    pub covered_hexes: Vec<CoveredHex>,
+    pub boosted_hex_eligibility: BoostedHexStatus,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("signal level {0:?} not allowed for {1:?}")]
+    InvalidSignalLevel(SignalLevel, RadioType),
+}
+
+pub fn calculate_coverage_points(radio: RewardableRadio) -> CoveragePoints {
+    let hex_coverage_points = radio.covered_hexes.calculated_coverage_points();
+    let location_trust_multiplier = radio.location_trust_scores.multiplier;
+    let speedtest_multiplier = radio.speedtests.multiplier;
+
+    let coverage_points = hex_coverage_points * location_trust_multiplier * speedtest_multiplier;
+    let total_coverage_points = coverage_points.round_dp_with_strategy(2, RoundingStrategy::ToZero);
+
+    CoveragePoints {
+        total_coverage_points,
+        hex_coverage_points,
+        location_trust_multiplier,
+        speedtest_multiplier,
+        // Radio information
```

Comment:
Yeah?
```suggestion
```

Reply:
Radio information has been removed from this struct entirely, in favor of allowing the caller to keep their `RewardableRadio`.